CRM/Contact/Import/Form/MapField.php

```diff
@@ -717,39 +717,15 @@ public function formatCustomFieldName(&$fields) {
    * @return \CRM_Contact_Import_Parser_Contact
    */
   public function submit($params, $mapperKeys) {
-    $mapper = $mapperKeysMain = array();
+    $mapper = $mapperKeysMain = $locations = array();
+    $parserParameters = CRM_Contact_Import_Parser_Contact::getParameterForParser($this->_columnCount);
 
     $phoneTypes = CRM_Core_PseudoConstant::get('CRM_Core_DAO_Phone', 'phone_type_id');
     $imProviders = CRM_Core_PseudoConstant::get('CRM_Core_DAO_IM', 'provider_id');
     $websiteTypes = CRM_Core_PseudoConstant::get('CRM_Core_DAO_Website', 'website_type_id');
     $locationTypes = CRM_Core_PseudoConstant::get('CRM_Core_DAO_Address', 'location_type_id');
 
-    //these mapper params need to set key as array and val as null.
-    $mapperParams = array(
-      'related' => 'relatedVal',
-      'locations' => 'locationsVal',
-      'mapperLocType' => 'mapperLocTypeVal',
-      'mapperPhoneType' => 'mapperPhoneTypeVal',
-      'mapperImProvider' => 'mapperImProviderVal',
-      'mapperWebsiteType' => 'mapperWebsiteTypeVal',
-      'relatedContactType' => 'relatedContactTypeVal',
-      'relatedContactDetails' => 'relatedContactDetailsVal',
-      'relatedContactLocType' => 'relatedContactLocTypeVal',
-      'relatedContactPhoneType' => 'relatedContactPhoneTypeVal',
-      'relatedContactImProvider' => 'relatedContactImProviderVal',
-      'relatedContactWebsiteType' => 'relatedContactWebsiteTypeVal',
-    );
-
-    //set respective mapper params to array.
-    foreach (array_keys($mapperParams) as $mapperParam) {
```

Comment:
This in conjunction with the $$mapperParam = NULL; loop does the same as CRM_Contact_Import_Parser_Contact::getParameterForParser